pinot-broker/src/main/java/org/apache/pinot/broker/routing/segmentpruner/interval/Interval.java

```diff
@@ -78,19 +74,17 @@ public int hashCode() {
     return result;
   }
 
+  @Nullable
   public static Interval getIntersection(Interval a, Interval b) {
-    Preconditions.checkNotNull(a, "Intersect invalid intervals {} and {}", a, b);
-    Preconditions.checkNotNull(b, "Intersect invalid intervals {} and {}", a, b);
     if (!a.intersects(b)) {
```

Comment:
There will be NPE if a is null.
Shall we create a static method in Interval, then do `Interval.intersects(a,b)`?

Reply:
It's okay to throw NPE here if there is a bug and we accidentally pass `null` into this method. Actually `Preconditions.checkNotNull()` also throws NPE. Here we just remove the extra check because this method could be called frequently. Added an assert to cache the `null` value in the test